### PR TITLE
Issue #1571

### LINKED ISSUE #1571: Terms can't be accessed when logged out on an incognito-enabled organization
That sounds complex, but it's very simple.

I believe this validation

https://github.com/mumuki/mumuki-laboratory/blob/f7981b5380c8b0f8e4c79e4a9c45df1cf2a628e8/app/views/users/terms.html.erb#L1-L7

is here so terms and conditions can be read even when logged out. The problem is `current_user?` is true for logged out users on incognito organizations:

https://github.com/mumuki/mumuki-laboratory/blob/f7981b5380c8b0f8e4c79e4a9c45df1cf2a628e8/lib/mumuki/laboratory/controllers/incognito_mode.rb#L9-L11

so visiting `organization/user/terms` fails when trying to get the user name:

```ruby
ActionView::Template::Error (undefined method `navigable_name'
for #<Mumuki::Domain::IncognitoClass:0x000055bfd3eb2768>):
    1: <%= content_for :breadcrumbs do %>
    2:   <% if current_user? %>
    3:     <%= breadcrumbs @user %>
    4:   <% else %>
    5:     <%= header_breadcrumbs %>
    6:   <% end %>
```

I'm thinking of two quick options:

* Having `t(:user)` as the navigable name of `IncognitoClass` and this won't happen again
* Remove the user from the terms breadcrumbs in all cases, though I understand it's there because `user` is part of the path and because those are the terms the user accepted

### PR DESCRIPTION
**Pull request by luchotc**
Resolves #1571 
Requires https://github.com/mumuki/mumuki-domain/pull/185

### CODE REVIEW

**Comment by luchotc** [comment]
Great finding @felipecalvo. I've ended up just changing the helper method that was used!

**Comment by luchotc** [comment]
This one solves another related issue that we weren't aware of. Incognito user wasn't working when trying to access forum routes. This was fixed in domain related PR

**Review by felipecalvo — APPROVED** [review]
I didn't know we had `current_logged_user?`, that's a better solution than what I proposed on the issue.

File: spec/features/terms_flow_spec.rb [hunk 1]
```diff
@@ -153,5 +153,35 @@
     end
   end
 
+  context 'with incognito mode' do
+    before { test_organization.update! incognito_mode_enabled: true }
```
Comment (flbulgarelli) [inline spec/features/terms_flow_spec.rb:157]: <3

**Review by flbulgarelli — APPROVED** [review]

**Pull request merged**